homeassistant/data_entry_flow.py

```diff
@@ -311,7 +321,60 @@ async def async_configure(
         if (
             data_schema := cur_step.get("data_schema")
         ) is not None and user_input is not None:
-            user_input = data_schema(user_input)
+            try:
+                user_input = data_schema(user_input)
+            except vol.Invalid as ex:
+                raised_errors = [ex]
+                if isinstance(ex, vol.MultipleInvalid):
+                    raised_errors = ex.errors
+
+                schema_errors: dict[str, Any] = {}
+                for error in raised_errors:
+                    if error.path:
+                        current_path_schema = data_schema.schema
+                        current_path_errors = schema_errors
+                        for path_part in error.path[:-1]:
+                            if path_part not in current_path_schema:
+                                current_path_schema = None
+                                break
+                            current_path_schema = current_path_schema[path_part]
+                            current_path_errors = current_path_errors.setdefault(
+                                path_part, {}
+                            )
+                        if (
+                            current_path_schema
+                            and error.path[-1] in current_path_schema
+                        ):
+                            current_path_errors[error.path[-1]] = error.error_message
+                            continue
+
+                    # If we get here, we could not identify the path of the error.
+                    schema_errors.get("base", []).append(str(error))
+
+                class SchemaInvalidError(
+                    InvalidData,
+                    ex.__class__,  # type: ignore[name-defined, misc]
+                ):
+                    """Error that contains schema errors."""
+
+                    def __getattribute__(self, name: str) -> Any:
+                        """Get attribute."""
+                        try:
+                            cause = super().__getattribute__("__cause__")
+                        except AttributeError:
+                            cause = None
+
+                        if cause:
+                            try:
+                                return getattr(cause, name)
+                            except AttributeError:
+                                pass
+
+                        return super().__getattribute__(name)
+
+                raise SchemaInvalidError(
```

Comment:
We raise now an exception instead of returning a Flowresult to behave the same as previously and therefore we don't need to fill out the schema with the input data